gig/api.py

```diff
@@ -0,0 +1,164 @@
+from typing import List, Optional
+
+from django.http import JsonResponse
+from ninja import Field, FilterSchema, ModelSchema, Query, Router, Schema
+
+from band.models import AssocStatusChoices
+from gig.models import Gig, Plan
+from gig.util import GigStatusChoices, PlanStatusChoices
+from member.models import Member
+
+
+class CannotResolveMemberStatus(Exception):
+    pass
+
+class CannotResolveGigStatus(Exception):
+    pass
+
+router = Router()
+
+class Message(Schema):
+    message: str
+
+class GigSchema(ModelSchema):
+    """
+        Gig schema for serialization. Resolves the plan status and gig status to human-readable strings. Asserts that response contains defined fields.
+    """
+    id: int
+    plan_status: str
+    band: Optional[str] = None
+    contact: Optional[str] = None
+    leader: Optional[str] = None
+    date: Optional[str] = None
+    call_time: Optional[str] = None
+    set_time: Optional[str] = None
+    end_time: Optional[str] = None
+    gig_status: Optional[str] = None
+
+    class Meta:
+        model = Gig
+        fields = [
+            "datenotes",
+            "title",
+            "details",
+            "setlist",
+            "dress",
+            "paid",
+            "postgig",
+            "is_full_day",
+            "address",
+            "is_archived",
+            "is_private",
+        ]
+
+    @staticmethod
+    def resolve_plan_status(obj, context):
+        plans = obj.member_plans
+        if plans:
+            auth_key = context.get("request").auth.get("key")
+            member = Member.objects.filter(api_key=auth_key).first()
+            if member:
+                plan = plans.filter(assoc__member=member).first()
+                if plan:
+                    try:
+                        return next((str(choice[1]) for choice in PlanStatusChoices.choices if choice[0] == plan.status))
+                    except StopIteration:
+                        raise CannotResolveMemberStatus
+        raise CannotResolveMemberStatus
+
+    @staticmethod
+    def resolve_band(obj):
+        return obj.band.name
+    
+    @staticmethod
+    def resolve_contact(obj):
+        return obj.contact.member_name
+    
+    @staticmethod
+    def resolve_leader(obj):
+        return obj.leader.name if obj.leader else obj.leader_text
+    
+    @staticmethod
+    def resolve_date(obj):
+        return obj.date.strftime("%Y-%m-%d") if obj.date else obj.setdate.strftime("%Y-%m-%d")
+    
+    @staticmethod
+    def resolve_call_time(obj):
+        return obj.date.strftime("%H:%M") if obj.date else obj.setdate.strftime("%H:%M")
```

Comment:
Are these both meant to point to the same date field? I see 4 date methods here, but Gigs only have 3 dates:
* `gig.date` which is the call time
* `gig.setdate` which is the start time
* `gig.enddate` which is when the gig is done

Reply:
And right after I posted that, I realized that one was date and one was time - you can ignore me